platforms/jvm/toolchains-jvm/src/main/java/org/gradle/jvm/toolchain/internal/install/JdkCacheDirectory.java

```diff
@@ -223,49 +247,34 @@ private static String getInstallFolderName(JvmInstallationMetadata metadata) {
         String os = OperatingSystem.current().getFamilyName();
         return String.format("%s-%s-%s-%s", vendor, version, architecture, os)
                 .replaceAll("[^a-zA-Z0-9\\-]", "_")
-                .toLowerCase();
+                .toLowerCase() + ".2";
     }
 
     private File unpack(File jdkArchive) {
         final FileTree fileTree = asFileTree(jdkArchive);
         String unpackFolderName = getNameWithoutExtension(jdkArchive);
-        final File unpackFolder = new File(jdkDirectory, unpackFolderName);
-        if (!unpackFolder.exists()) {
-            operations.copy(spec -> {
-                spec.from(fileTree);
-                spec.into(unpackFolder);
-                spec.setDuplicatesStrategy(DuplicatesStrategy.WARN);
-            });
-        }
+        final File unpackFolder = temporaryFileProvider.createTemporaryDirectory(unpackFolderName, null, "jdks");
+        unpackFolder.deleteOnExit();
```

Comment:
💭 Don't we already sort of do this in a finally block? Or is this just a further just-in-case protection against leaving the unpack directory behind?

Reply:
This is an attempt to ensure that it's always deleted, even in the event of sudden termination without unwind, for example SIGTERM:
> The Java virtual machine shuts down in response to two kinds of events:
> - The program exits normally, when the last non-daemon thread exits or when the [exit](https://docs.oracle.com/javase/8/docs/api/java/lang/Runtime.html#exit-int-) (equivalently, [System.exit](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#exit-int-)) method is invoked, or
> - The virtual machine is terminated in response to a user interrupt, such as typing ^C, or a system-wide event, such as user logoff or system shutdown.

\- [`Runtime.addShutdownHook`](https://docs.oracle.com/javase/8/docs/api/java/lang/Runtime.html#addShutdownHook-java.lang.Thread-) docs